config.cpp

```diff
@@ -41,33 +44,44 @@ Registry::Registry()
 	);
 }
 
+Registry::~Registry() {
+	RegCloseKey(m_reg_key);
+}
+
 Config Registry::get_config() {
+	Registry registry;
 	Config config;
 
 	const std::wstring is_registry_migrated_name = L"IsRegistryMigrated";
-	bool is_registry_migrated = get(is_registry_migrated_name, 0.0f) != 0.0f;
+	bool is_registry_migrated = registry.get(is_registry_migrated_name, 0.0f) != 0.0f;
 
 	for (const auto &opt : Cfg::All) {
-		if (is_registry_migrated) {
-			config[opt] = get(opt.name, opt.default_);
+		bool has_no_legacy = opt.legacy_id == 0;
+		if (is_registry_migrated || has_no_legacy) {
+			config[opt] = registry.get(opt.name, opt.default_);
 		} else {
-			config[opt] = get(std::to_wstring(opt.legacy_id), opt.default_);
-			write(opt.name, config[opt]);
-			remove(std::to_wstring(opt.legacy_id));
+			config[opt] = registry.get(std::to_wstring(opt.legacy_id), opt.default_);
+			registry.write(opt.name, config[opt]);
+			registry.remove(std::to_wstring(opt.legacy_id));
 		}
 	}
 
 	if (!is_registry_migrated) {
-		write(is_registry_migrated_name, 1.0f);
+		registry.write(is_registry_migrated_name, 1.0f);
 	}
 
 	return config;
 }
 
 float Registry::get(const std::wstring &opt, float default_) {
-	wchar_t result[1 << 6] {};
+	auto value = get_string(opt, std::to_wstring(default_));
+	return std::stof(value);
+}
+
+std::wstring Registry::get_string(const std::wstring &opt, const std::wstring &default_) {
+	wchar_t result[1 << 12] {};
```

Comment:
I'm too tired to fix this right now but this looks like a free stack smash to me... this needs to get looked at again.